File: ALGORITMOS-E-ESTRUTURAS-DE-DADOS-II (PT2)/#Teorico/Códigos de Estudos/3 - Estruturas de Dados Lineares e Flexíveis/src/Fila.java
public class Fila {
    int frente;
    int tras;
    int[] vetor;
    int n;

    Fila(int n) {
        this.frente = 0;
        this.tras = 0;
        this.n = n;
        this.vetor = new int[n];
    }

    void enfileirar(int item) {
        if (this.isCheia() == false) {
            this.vetor[tras] = item;
            this.tras = (this.tras + 1) % this.n;
        }
    }

    int desenfileirar() {
        if (this.isVazia() == false) {
            int valor = this.vetor[this.frente];
            this.frente = (this.frente + 1) % this.n;
            return valor;
        }
        return -1;
    }

    boolean isVazia() {
        if (this.tras == this.frente) {
            System.out.println("Fila vazia");
            return true;
        }
        return false;
    }

    boolean isCheia() {
        if (((this.tras + 1) % this.n) == this.frente) {
            System.out.println("Fila cheia");
            return true;
        }
        return false;
    }

    void imprimir() {
        int ini = this.frente;
        int fim = this.tras;
        while (ini != fim) {
            System.out.print(vetor[ini] + ", ");
            ini = (ini + 1) % this.n;
        }
    }

    public static void main(String[] args) {
        Fila fila = new Fila(10);
        fila.enfileirar(3);
        fila.enfileirar(5);
        fila.enfileirar(1);
        fila.enfileirar(2);
        fila.enfileirar(4);
        fila.imprimir();

        System.out.println();
        fila.desenfileirar();
        fila.desenfileirar();
        fila.imprimir();

        System.out.println();
        fila.desenfileirar();
        fila.desenfileirar();
        fila.desenfileirar();
        fila.desenfileirar();

        for (int i = 0; i < 9; i++) {
            fila.enfileirar(i * 2);
        }
        fila.imprimir();
        fila.enfileirar(18);
    }
}

File: ALGORITMOS-E-ESTRUTURAS-DE-DADOS-II (PT3)/Algoritmos e Estruturas de Dados II - TEO/TAD - Tipo Abstrato de Dados/FilaAutoReferenciada/FilaAutoReferenciada/Fila.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace FilaAutoReferenciada{
	class Fila{
		private Celula frente;
		private Celula tras;

		public Fila(){

			Celula sentinela;

			sentinela = new Celula();
			frente = sentinela;
			tras = sentinela;
		}

		public Boolean filaVazia(){

			Boolean resp;

			if (frente == tras)
				resp = true;
			else
				resp = false;

			return resp;
		}

		public void enfileirar(Inteiro novo){

			Celula novaCelula;

			novaCelula = new Celula(novo);

			tras.Proximo = novaCelula;
			tras = novaCelula;
		}

		public Inteiro desenfileirar(){

			Celula celulaDesenfileirada;
			Celula proximaCelula;

			if (!filaVazia()){

				celulaDesenfileirada = frente.Proximo;
				proximaCelula = celulaDesenfileirada.Proximo;
				frente.Proximo = proximaCelula;

				if (celulaDesenfileirada == tras) // se a célula retirada for a última célula da fila
					tras = frente;

				return (celulaDesenfileirada.Item);
			}
			else
				throw new Exception("Não foi possível desenfileirar nenhum item: a fila está vazia!");
		}

		public void imprimir(){

			Celula aux;

			if (!filaVazia()){
				Console.WriteLine("Conteúdo da fila:");

				aux = frente.Proximo;

				while (aux != null){
					aux.Item.imprimir();
					aux = aux.Proximo;
				}
			}
			else
				throw new Exception("Não foi possível imprimir o conteúdo da fila: a fila está vazia!");
		}
	}
}
